pkg/pkgrepository/app_deploy.go

```diff
@@ -30,13 +30,105 @@ func (a *App) deploy(tplOutput string) exec.CmdRunResult {
 			return exec.NewCmdRunResultWithErr(fmt.Errorf("Preparing kapp: %s", err))
 		}
 
-		return kapp.Deploy(tplOutput, a.startFlushingAllStatusUpdates, func(exec.CmdRunResult) {})
+		// TODO: should we inject this somewhere else? - there's some other rules that might interact with it...
```

Comment:
@cppforlife   i left this in the same spot as the prototype for now but curious what your thoughts are --  iirc there's a trick that it can't be inserted too early so it's slightly delicate?

Reply:
add where we already add other stuff -- https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/pkg/pkgrepository/package_repo_app.go#L82